pulsar-broker/src/test/java/org/apache/pulsar/schema/SchemaTest.java

```diff
@@ -858,36 +862,62 @@ private void checkSchemaForAutoSchema(int index, Message<GenericRecord> message)
             Assert.fail("Failed to get reader schema for auto consume multiple schema topic.");
         }
         Object nativeObject = message.getValue().getNativeObject();
+        String baseTopic = message.getProperty("baseTopic");
         JsonNode jsonNode;
-        switch (index) {
-            case 0:
+        KeyValue kv;
+        switch (baseTopic) {
+            case "bytes_schema":
                 Assert.assertEquals(new String((byte[]) nativeObject), "bytes value");
                 break;
-            case 1:
+            case "string_schema":
                 Assert.assertEquals((String) nativeObject, "string value");
                 break;
-            case 2:
+            case "bool_schema":
                 Assert.assertEquals(nativeObject, Boolean.TRUE);
                 break;
-            case 3:
+            case "json_one_schema":
                 jsonNode = (JsonNode) nativeObject;
                 Assert.assertEquals(jsonNode.get("id").intValue(), 1);
                 break;
-            case 4:
+            case "json_three_schema":
                 jsonNode = (JsonNode) nativeObject;
                 Assert.assertEquals(jsonNode.get("id").intValue(), 3);
                 Assert.assertEquals(jsonNode.get("name").textValue(), "ran");
                 break;
-            case 5:
+            case "json_four_schema":
                 jsonNode = (JsonNode) nativeObject;
                 Assert.assertEquals(jsonNode.get("id").intValue(), 4);
                 Assert.assertEquals(jsonNode.get("name").textValue(), "tang");
                 Assert.assertEquals(jsonNode.get("age").intValue(), 18);
                 break;
-            case 6:
+            case "avro_one_schema":
                 org.apache.avro.generic.GenericRecord genericRecord =
                         (org.apache.avro.generic.GenericRecord) nativeObject;
-                Assert.assertEquals(genericRecord.get("id"), 1);
+                Assert.assertEquals(genericRecord.get("id"), 10);
+                break;
+            case "k_one_v_three_schema_separate":
+                kv = (KeyValue<?, ?>) nativeObject;
```

Comment:
this should be a KeyValue<GenericRecord, GenericRecord>

Reply:
Ok, I'll fix.